src/theory/bv/theory_bv.cpp

```diff
@@ -622,17 +622,18 @@ int TheoryBV::getReduction(int effort, Node n, Node& nr)
     const unsigned size = utils::getSize(n[0]);
     const Node z = nm->mkConst(Rational(0));
     const Node bvone = utils::mkOne(1);
-    NodeBuilder<> result(kind::PLUS);
+    std::vector<Node> children;
     Integer i = 1;
     for (unsigned bit = 0; bit < size; ++bit, i *= 2)
     {
       Node cond =
           nm->mkNode(kind::EQUAL,
                      nm->mkNode(nm->mkConst(BitVectorExtract(bit, bit)), n[0]),
                      bvone);
-      result << nm->mkNode(kind::ITE, cond, nm->mkConst(Rational(i)), z);
+      children.push_back(
+          nm->mkNode(kind::ITE, cond, nm->mkConst(Rational(i)), z));
     }
-    nr = Node(result);
+    nr = children.size() == 1 ? children[0] : nm->mkNode(kind::PLUS, children);
```

Comment:
How about using `RewriteRule<BVToNatEliminate>::apply` instead of duplicating the code?

Reply:
That function is private so I've added a bit-vector utility that both call now.